src/binarySearch.js

```diff
@@ -0,0 +1,22 @@
+export default function binarySearch(array, number) {
+//  your code here
+const length = array.length
+const middle = Math.ceil(array.length / 2)
+
+  if (number == array[middle]) {
+    return array[middle], number
+  }
+  else if (number > array[middle]) {
+    let newArray = array.slice(0, array[middle])
+    binarySearch(newArray)
+
+    return array[middle], number
+  }
+  else if (number < array[middle]) {
+    let newArray = array.slice(array[middle])
+    binarySearch(newArray)
+
+    return array[i], number
+
+  }
+}
```

Comment:
I feel you need some kind of while or for loop to do the iterations you need. The three conditionals (the if/elses) make sense as a binary search pattern but `i` is not defined and it cannot iterate in this iteration.